cunumeric/fft/fft.py

```diff
@@ -0,0 +1,857 @@
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from cunumeric.array import ndarray
+from cunumeric.module import add_boilerplate
+from cunumeric.config import FFTCode, FFTDirection, FFTNormalization
+
+
+def _sanitize_user_axes(a, s, axes, is_c2r=False):
+    if s is None:
+        user_shape = False
+        if axes is None:
+            s = list(a.shape)
+        else:
+            s = [a.shape[ax] for ax in axes]
+    else:
+        user_shape = True
+        s = list(s)
+    if axes is None:
+        axes = list(range(-len(s), 0))
```

Comment:
Why not just `list(range(len(s)))`? Indexing-wise, they should be equivalent. Unless the axis have to be negative?